### PR TITLE
New integration test helper class that makes tests truly independent

### PR DESCRIPTION
**Pull request by dolgopolovwork**
New helper class for making independent integration tests. Also roles were ranamed, because it can not contain uppercase letters.

### CODE REVIEW

File: src/main/kotlin/registration/EthFreeWalletsProvider.kt [hunk 1]
```diff
@@ -16,29 +15,33 @@ import sidechain.iroha.util.getRelays
...
      */
     fun getWallet(): Result<String, Exception> {
+        return getWallets().map { freeWallets -> freeWallets.first() }
+    }
+
+    /**
+     * Get all free ethereum relay wallets
+     * @return free ethereum relay wallets
+     */
+    fun getWallets(): Result<Set<String>, Exception> {
```
Comment (Alexey-N-Chernyshov) [inline src/main/kotlin/registration/EthFreeWalletsProvider.kt]: may be rename to getFreeRelays()?

File: src/main/kotlin/notary/EthWalletsProviderIrohaImpl.kt [hunk 2]
```diff
@@ -13,15 +13,15 @@ import sidechain.iroha.util.getRelays
...
-    val irohaConfig: IrohaConfig,
-    val keypair: Keypair,
-    val irohaNetwork: IrohaNetwork,
-    val account: String,
-    val setter: String
+    private val irohaConfig: IrohaConfig,
+    private val keypair: Keypair,
+    private val irohaNetwork: IrohaNetwork,
+    private val notaryIrohaAccount: String,
+    private val registrationAccount: String
```
Comment (Alexey-N-Chernyshov) [inline src/main/kotlin/notary/EthWalletsProviderIrohaImpl.kt]: In general case, account details can be set not by `registrationAccount`. I think we need more meaningful name, like 'accountDetailSetter' and 'accountDetailHolder'.

File: src/main/kotlin/registration/EthFreeWalletsProvider.kt (hunk 1 above)
Comment (dolgopolovwork) [reply src/main/kotlin/registration/EthFreeWalletsProvider.kt]: what about renaming the whole class as well?

File: src/main/kotlin/notary/EthWalletsProviderIrohaImpl.kt (hunk 2 above)
Comment (dolgopolovwork) [reply src/main/kotlin/notary/EthWalletsProviderIrohaImpl.kt]: yes that is true. but this is wallets provider class and the only account that can add new wallet to Iroha is registrationAccount.

File: src/main/kotlin/registration/EthFreeWalletsProvider.kt (hunk 1 above)
Comment (Alexey-N-Chernyshov) [reply src/main/kotlin/registration/EthFreeWalletsProvider.kt]: EthFreeRelayProvider

**Pull request merged**